client/resources/pixel-buffer.cpp

```diff
@@ -6,219 +6,192 @@
-					if (b < 3) std::fputc(p[i], stderr); // only output RGB
-				}
-				break;
-			case ImageColorMode::MONOCHROME:
-			case ImageColorMode::MONOCHROME_A:
-			case ImageColorMode::UNKNOWN_MODE:
-				break;
+void PixelBuffer::PPMDebug() {
+	// output a PPM image to stderr as a debug feature
+	std::fprintf(stderr, "P6\n%d\n%d\n%d\n", imagewidth, imageheight, (1u << imagebitdepth) - 1);
+	if (!blockptr)
+		return;
+	uint8_t* p = blockptr.get();
+	switch (imagemode) {
+	case ImageColorMode::COLOR_RGB:
+		for (uint32_t i = 0; i < bufferpitch * imageheight; i++) {
+			std::fputc(p[i], stderr);
 		}
-	}
-	void PixelBuffer::PPMDebug(const char * ofile) {
-		// output a PPM image to stderr as a debug feature
-		std::fstream file(ofile, std::ios::out | std::ios::binary);
-		if (!file.is_open()) {
-			return;
-		}
-		file << "P6\n" << imagewidth << '\n' << imageheight << '\n';
-		file << ((1u << imagebitdepth) - 1) << '\n';
-		if (!blockptr) return;
-		uint8_t * p = blockptr.get();
-		switch (imagemode) {
-			case ImageColorMode::COLOR_RGB:
-				file.write((char*)p, bufferpitch * imageheight);
-				break;
-			case ImageColorMode::COLOR_RGBA:
-				for (uint32_t i = 0; i < bufferpitch * imageheight; i += 4) {
-					file.write((char*)(p + i), 3); // only output RGB
-				}
-				break;
-			case ImageColorMode::MONOCHROME:
-			case ImageColorMode::MONOCHROME_A:
-			case ImageColorMode::UNKNOWN_MODE:
-				break;
-
+		break;
+	case ImageColorMode::COLOR_RGBA:
+		for (uint32_t b = 0, i = 0; i < bufferpitch * imageheight; i++, b++) {
+			if (b == 4)
+				b = 0;
+			if (b < 3)
+				std::fputc(p[i], stderr); // only output RGB
 		}
+		break;
+	case ImageColorMode::MONOCHROME:
+	case ImageColorMode::MONOCHROME_A:
+	case ImageColorMode::UNKNOWN_MODE: break;
 	}
-
-	bool PixelBuffer::Create(uint32_t width, uint32_t height, uint32_t bitspersample, ImageColorMode mode) {
-		uint8_t pixelsize;
-		switch (mode) {
-			case ImageColorMode::COLOR_RGB:
-				pixelsize = 3;
-				break;
-			case ImageColorMode::COLOR_RGBA:
-				pixelsize = 4;
-				break;
-			case ImageColorMode::MONOCHROME:
-				pixelsize = 1;
-				break;
-			case ImageColorMode::MONOCHROME_A:
-				pixelsize = 2;
-				break;
-			default:
-				return false;
-		}
-		writelock.lock();
-		imagepixelsize = pixelsize;
-		imagewidth = width;
-		imageheight = height;
-		imagebitdepth = bitspersample;
-		imagemode = mode;
-		image_x = 0;
-		image_y = 0;
-		dirty = true;
-		bufferpitch = (bitspersample * imagepixelsize);
-		bufferpitch = width * ((bufferpitch >> 3u) + ((bufferpitch & 0x7u) ? 1 : 0));
-
-		blockptr = std::unique_ptr<uint8_t[]>(new uint8_t[bufferpitch * height]);
-		if (!blockptr) {
-			writelock.unlock();
-			return false;
-		}
-		std::memset(blockptr.get(), 0, bufferpitch * height);
-		writelock.unlock();
-		return true;
+}
+void PixelBuffer::PPMDebug(const char* ofile) {
+	// output a PPM image to stderr as a debug feature
+	std::fstream file(ofile, std::ios::out | std::ios::binary);
```

Comment:
comment copy pasta, this overload outputs to a file, not stderr

Reply:
Changed to `// output a PPM image to file`